### ISSUE TITLE
Allow to define proxy settings for the P2 resolver

### ISSUE THREAD

**Issue by silmarx** [reficio/p2-maven-plugin#76]
When running the example of the P2 resolver I was not able to access to the P2 repository even the proxy was defined in the maven settings file. I suppose the P2 resolver is not using the maven settings. If I run the example at home where there's no proxy the example runs fine.

My suggestion is to reuse the setting from Maven or allow to define a proxy for the p2-maven-plugin, otherwise this can't be used behind a proxy, which is very a very common scenario in enterprise environments.

**Comment by leaobr** [comment]
This is definitely a blocker. Can anyone else confirm that it doesn't read the maven proxy settings?

**Comment by thraidh** [comment]
Yes, I can confirm this. Using the same repos with tycho works, but `p2-maven-plugin` definitely wants to connect directly.

**Comment by thraidh** [comment]
You can work around the problem by using `mvn -Dhttp.proxyHost=proxy.your-company.com -Dhttp.proxyPort=8888`. Still, it would be better if the files were downloaded using the proxy specified maven settings.
